kyo-prelude/shared/src/main/scala/kyo/Stream.scala

```diff
@@ -168,18 +159,18 @@ sealed abstract class Stream[V, -S]:
       *   A new stream containing at most n elements from the original stream
       */
     def take(n: Int)(using tag: Tag[Emit[Chunk[V]]], frame: Frame): Stream[V, S] =
-        if n <= 0 then discard
+        val kyoUnit = Kyo.pure[Unit, S & Emit[Chunk[V]]](())
```

Comment:
can this be moved to the `Emit.valueWith` function or is there an inference issue?